### DISCUSSION TITLE
Problem with stacks

### DISCUSSION THREAD

**Discussion by tdelov** [photoprism/photoprism#3248] [Q&A]
I have found that images are getting stacked with the same base filename, even if they are in different folders.

For example these two files are incorrectly getting stacked.

```
Filename | iphonebackup/2019/IMG_1040.JPG
Filename | iphonebackup/2023/IMG_1040.MOV
```

As you can see, they are from different folders (sorted by year) and not related in any other way other than they share the base name.

I can not seem to unstack them manually. Even if I could, I have around 70,000 photos, so it might not be achievable to go through the whole collection to find images not correctly stacked.

Is there a way I could force files from different folders not to be stacked?

**Comment by lastzero** [comment]
Please check all the metadata using exiftool first. Could they have a common unique ID, for example added by a software bug?

**Comment by tdelov** [comment]
I can not see a common ID.
They were originally in the  same iphonebackup folder, I then moved them into folders sorted by year and re-index the collection.
`podman exec photoprism photoprism index --force --cleanup`

Photoprisum still stacked them together. 
Perhaps there are some steps I needed to do?

It seems to be part of the expected behavior according to the documentation that they are stacked if in the same location. I think.
https://docs.photoprism.app/user-guide/settings/library/

> Files that have exactly the same file and folder name are always stacked, for example /2018/IMG_1234.jpg and /2018/IMG_1234.avi.
> 
> When you change the stacking settings, already stacked files will not be unstacked automatically.
> 



```
exiftool 2019/IMG_1040.JPG
ExifTool Version Number         : 12.42
File Name                       : IMG_1040.JPG
Directory                       : 2019
File Size                       : 197 kB
File Modification Date/Time     : 2019:02:25 02:17:43+11:00
File Access Date/Time           : 2023:03:06 13:40:45+11:00
File Inode Change Date/Time     : 2019:02:25 02:17:43+11:00
File Permissions                : -rwxr-xr-x
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
JFIF Version                    : 1.01
Exif Byte Order                 : Big-endian (Motorola, MM)
Orientation                     : Horizontal (normal)
X Resolution                    : 72
Y Resolution                    : 72
Resolution Unit                 : inches
Y Cb Cr Positioning             : Centered
Exif Version                    : 0221
Components Configuration        : Y, Cb, Cr, -
Flashpix Version                : 0100
Color Space                     : sRGB
Exif Image Width                : 960
Exif Image Height               : 1280
Scene Capture Type              : Standard
Compression                     : JPEG (old-style)
Thumbnail Offset                : 316
Thumbnail Length                : 9097
Current IPTC Digest             : d41d8cd98f00b204e9800998ecf8427e
IPTC Digest                     : d41d8cd98f00b204e9800998ecf8427e
Image Width                     : 960
Image Height                    : 1280
Encoding Process                : Baseline DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 960x1280
Megapixels                      : 1.2
Thumbnail Image                 : (Binary data 9097 bytes, use -b option to extract)

```


```
exiftool 2023/IMG_1040.MOV 
ExifTool Version Number         : 12.42
File Name                       : IMG_1040.MOV
Directory                       : 2023
File Size                       : 14 MB
File Modification Date/Time     : 2023:02:25 10:51:33+11:00
File Access Date/Time           : 2023:03:06 15:29:07+11:00
File Inode Change Date/Time     : 2023:02:25 10:51:33+11:00
File Permissions                : -rwxr-xr-x
File Type                       : MOV
File Type Extension             : mov
MIME Type                       : video/quicktime
Major Brand                     : Apple QuickTime (.MOV/QT)
Minor Version                   : 0.0.0
Compatible Brands               : qt
Media Data Size                 : 14042491
Media Data Offset               : 36
Movie Header Version            : 0
Create Date                     : 2023:02:24 23:51:33
Modify Date                     : 2023:02:24 23:51:43
Time Scale                      : 600
Duration                        : 9.17 s
Preferred Rate                  : 1
Preferred Volume                : 100.00%
Preview Time                    : 0 s
Preview Duration                : 0 s
Poster Time                     : 0 s
Selection Time                  : 0 s
Selection Duration              : 0 s
Current Time                    : 0 s
Next Track ID                   : 6
Track Header Version            : 0
Track Create Date               : 2023:02:24 23:51:33
Track Modify Date               : 2023:02:24 23:51:43
Track ID                        : 1
Track Duration                  : 9.17 s
Track Layer                     : 0
Track Volume                    : 0.00%
Image Width                     : 1920
Image Height                    : 1080
Clean Aperture Dimensions       : 1920x1080
Production Aperture Dimensions  : 1920x1080
Encoded Pixels Dimensions       : 1920x1080
Graphics Mode                   : ditherCopy
Op Color                        : 32768 32768 32768
Compressor ID                   : hvc1
Source Image Width              : 1920
Source Image Height             : 1080
X Resolution                    : 72
Y Resolution                    : 72
Compressor Name                 : HEVC
Bit Depth                       : 24
Video Frame Rate                : 59.967
Balance                         : 0
Audio Format                    : mp4a
Audio Channels                  : 2
Audio Bits Per Sample           : 16
Audio Sample Rate               : 44100
Purchase File Format            : mp4a
Warning                         : [minor] The ExtractEmbedded option may find more tags in the media data
Matrix Structure                : 1 0 0 0 1 0 0 0 1
Content Describes               : Track 1
Media Header Version            : 0
Media Create Date               : 2023:02:24 23:51:33
Media Modify Date               : 2023:02:24 23:51:43
Media Time Scale                : 600
Media Duration                  : 9.17 s
Media Language Code             : und
Gen Media Version               : 0
Gen Flags                       : 0 0 0
Gen Graphics Mode               : ditherCopy
Gen Op Color                    : 32768 32768 32768
Gen Balance                     : 0
Handler Class                   : Data Handler
Handler Vendor ID               : Apple
Handler Description             : Core Media Data Handler
Meta Format                     : mebx
Handler Type                    : Metadata Tags
Location Accuracy Horizontal    : 17.755240
GPS Coordinates                 : 38 deg 8' 16.80" S, 144 deg 20' 53.16" E, 23.322 m Above Sea Level
Make                            : Apple
Model                           : iPhone 11
Software                        : 16.3
Creation Date                   : 2023:02:25 10:51:33+11:00
Image Size                      : 1920x1080
Megapixels                      : 2.1
Avg Bitrate                     : 12.2 Mbps
GPS Altitude                    : 23.322 m
GPS Altitude Ref                : Above Sea Level
GPS Latitude                    : 38 deg 8' 16.80" S
GPS Longitude                   : 144 deg 20' 53.16" E
Rotation                        : 180
GPS Position                    : 38 deg 8' 16.80" S, 144 deg 20' 53.16" E

```

**Reply by lastzero** [reply]
As documented, already stacked files won't get unstacked by reindexing.

**Comment by tdelov** [comment]
So what should I to do?

**Comment by ALEEF02** [comment]
I am having an extremely similar issue. Images and videos with the same name (but different extensions, obviously) are getting stacked, regardless of the fact that they may be years apart. Are there any updates about this issue?

**Reply by lastzero** [reply]
What update? We can remove duplicate detection and stacks.

**Reply by lastzero** [reply]
Kindly note that there can be edited versions of originals years after they were taken. The easiest way to prevent this us to simply use another filename or another folder.

**Reply by ALEEF02** [reply]
Ah, but see I am using different file names. For example, `2023-05-02_21-33-15_IMG_6177.JPG` is getting stacked onto `2021-07-16_12-43-54_IMG_6177.HEIC`, even though the ONLY bit that matches between these files is the `IMG_6177`. The metadata times, locations, titles, and filenames's date are all different (see photo).
![image](https://github.com/photoprism/photoprism/assets/12904259/63f22f69-8582-4822-acf9-3b10a4e5d8fb)

**Reply by lastzero** [reply]
Did you disable all the other stacking options BEFORE that happened? https://docs.photoprism.app/user-guide/settings/library/#stacks

**Reply by ALEEF02** [reply]
No. `Place & Time` and `Unique ID` were still on. These shouldn't cause them to be stacked. I should also mention that the JPG of the person was created from a .MOV with the same original filename `2023-05-02_21-33-15_IMG_6177.MOV`, so there are 4 files in this automatically created stack.

**Reply by graciousgrey** [reply]
> No. Place & Time and Unique ID were still on

You can use `exiftool -j filename` to check the metadata of the original files. Maybe they share the same InstanceID or DocumentID.

**Reply by ALEEF02** [reply]
Hi @graciousgrey. Life got busy, yet I am still having this problem. `2021-07-16_12-43-54_IMG_6177.HEIC` has the following metadata:
```json
[{
  "SourceFile": "originals/2021/07/20210716_194354_46773667.heic",
  "ExifToolVersion": 12.56,
  "FileName": "20210716_194354_46773667.heic",
  "Directory": "originals/2021/07",
  "FileSize": "1599 kB",
  "FileModifyDate": "2023:08:27 01:48:17-07:00",
  "FileAccessDate": "2023:08:27 01:48:17-07:00",
  "FileInodeChangeDate": "2023:08:27 16:10:16-07:00",
  "FilePermissions": "-rw-rw-r--",
  "FileType": "HEIC",
  "FileTypeExtension": "heic",
  "MIMEType": "image/heic",
  "MajorBrand": "High Efficiency Image Format HEVC still image (.HEIC)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["mif1","MiPr","miaf","MiHB","heic"],
  "HandlerType": "Picture",
  "PrimaryItemReference": 49,
  "MetaImageSize": "4032x3024",
  "ExifByteOrder": "Big-endian (Motorola, MM)",
  "Make": "Apple",
  "Model": "iPhone 11 Pro",
  "Orientation": "Horizontal (normal)",
  "XResolution": 72,
  "YResolution": 72,
  "ResolutionUnit": "inches",
  "Software": 14.6,
  "ModifyDate": "2021:07:16 12:43:54",
  "HostComputer": "iPhone 11 Pro",
  "YCbCrPositioning": "Centered",
  "ExposureTime": "1/60",
  "FNumber": 1.8,
  "ExposureProgram": "Program AE",
  "ISO": 400,
  "ExifVersion": "0232",
  "DateTimeOriginal": "2021:07:16 12:43:54",
  "CreateDate": "2021:07:16 12:43:54",
  "OffsetTime": "-07:00",
  "OffsetTimeOriginal": "-07:00",
  "OffsetTimeDigitized": "-07:00",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "ShutterSpeedValue": "1/60",
  "ApertureValue": 1.8,
  "BrightnessValue": 1.019331428,
  "ExposureCompensation": 0,
  "MeteringMode": "Multi-segment",
  "Flash": "Off, Did not fire",
  "FocalLength": "4.2 mm",
  "SubjectArea": "2002 1505 2213 1324",
  "MakerNoteVersion": 12,
  "RunTimeFlags": "Valid",
  "RunTimeValue": 611776366626041,
  "RunTimeScale": 1000000000,
  "RunTimeEpoch": 0,
  "AEStable": "No",
  "AETarget": 173,
  "AEAverage": 163,
  "AFStable": "Yes",
  "AccelerationVector": "-1.014162421 0.001299376832 0.05679097028",
  "FocusDistanceRange": "1.21 - 1.46 m",
  "LivePhotoVideoIndex": 575676416,
  "HDRHeadroom": 0.7844365247,
  "SignalToNoiseRatio": 31.54694557,
  "FocusPosition": 127,
  "SubSecTimeOriginal": 629,
  "SubSecTimeDigitized": 629,
  "FlashpixVersion": "0100",
  "ColorSpace": "Uncalibrated",
  "ExifImageWidth": 4032,
  "ExifImageHeight": 3024,
  "SensingMethod": "One-chip color area",
  "SceneType": "Directly photographed",
  "ExposureMode": "Auto",
  "WhiteBalance": "Auto",
  "FocalLengthIn35mmFormat": "26 mm",
  "SceneCaptureType": "Standard",
  "LensInfo": "1.539999962-6mm f/1.8-2.4",
  "LensMake": "Apple",
  "LensModel": "iPhone 11 Pro back triple camera 4.25mm f/1.8",
  "CompositeImage": "General Composite Image",
  "GPSLatitudeRef": "North",
  "GPSLongitudeRef": "West",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSSpeedRef": "km/h",
  "GPSSpeed": 0,
  "GPSImgDirectionRef": "Magnetic North",
  "GPSImgDirection": REDACTED,
  "GPSDestBearingRef": "Magnetic North",
  "GPSDestBearing": REDACTED,
  "GPSDateStamp": "2021:07:16",
  "GPSHPositioningError": "1414 m",
  "ProfileCMMType": "Apple Computer Inc.",
  "ProfileVersion": "4.0.0",
  "ProfileClass": "Display Device Profile",
  "ColorSpaceData": "RGB ",
  "ProfileConnectionSpace": "XYZ ",
  "ProfileDateTime": "2017:07:07 13:22:32",
  "ProfileFileSignature": "acsp",
  "PrimaryPlatform": "Apple Computer Inc.",
  "CMMFlags": "Not Embedded, Independent",
  "DeviceManufacturer": "Apple Computer Inc.",
  "DeviceModel": "",
  "DeviceAttributes": "Reflective, Glossy, Positive, Color",
  "RenderingIntent": "Perceptual",
  "ConnectionSpaceIlluminant": "0.9642 1 0.82491",
  "ProfileCreator": "Apple Computer Inc.",
  "ProfileID": "ca1a9582257f104d389913d5d1ea1582",
  "ProfileDescription": "Display P3",
  "ProfileCopyright": "Copyright Apple Inc., 2017",
  "MediaWhitePoint": "0.95045 1 1.08905",
  "RedMatrixColumn": "0.51512 0.2412 -0.00105",
  "GreenMatrixColumn": "0.29198 0.69225 0.04189",
  "BlueMatrixColumn": "0.1571 0.06657 0.78407",
  "RedTRC": "(Binary data 32 bytes, use -b option to extract)",
  "ChromaticAdaptation": "1.04788 0.02292 -0.0502 0.02959 0.99048 -0.01706 -0.00923 0.01508 0.75168",
  "BlueTRC": "(Binary data 32 bytes, use -b option to extract)",
  "GreenTRC": "(Binary data 32 bytes, use -b option to extract)",
  "HEVCConfigurationVersion": 1,
  "GeneralProfileSpace": "Conforming",
  "GeneralTierFlag": "Main Tier",
  "GeneralProfileIDC": "Main Still Picture",
  "GenProfileCompatibilityFlags": "Main Still Picture, Main 10, Main",
  "ConstraintIndicatorFlags": "176 0 0 0 0 0",
  "GeneralLevelIDC": "90 (level 3.0)",
  "MinSpatialSegmentationIDC": 0,
  "ParallelismType": 0,
  "ChromaFormat": "4:2:0",
  "BitDepthLuma": 8,
  "BitDepthChroma": 8,
  "AverageFrameRate": 0,
  "ConstantFrameRate": "Unknown",
  "NumTemporalLayers": 1,
  "TemporalIDNested": "No",
  "ImageWidth": 4032,
  "ImageHeight": 3024,
  "ImageSpatialExtent": "4032x3024",
  "Rotation": 0,
  "ImagePixelDepth": "8 8 8",
  "MediaDataSize": 1595044,
  "MediaDataOffset": 4366,
  "RunTimeSincePowerUp": "7 days 1:56:16",
  "Aperture": 1.8,
  "ImageSize": "4032x3024",
  "Megapixels": 12.2,
  "ScaleFactor35efl": 6.1,
  "ShutterSpeed": "1/60",
  "SubSecCreateDate": "2021:07:16 12:43:54.629-07:00",
  "SubSecDateTimeOriginal": "2021:07:16 12:43:54.629-07:00",
  "SubSecModifyDate": "2021:07:16 12:43:54-07:00",
  "GPSAltitude": "REDACTED",
  "GPSLatitude": "REDACTED",
  "GPSLongitude": "REDACTED",
  "CircleOfConfusion": "0.005 mm",
  "FOV": "69.4 deg",
  "FocalLength35efl": "4.2 mm (35 mm equivalent: 26.0 mm)",
  "GPSPosition": "REDACTED",
  "HyperfocalDistance": "2.04 m",
  "LightValue": 5.6,
  "LensID": "iPhone 11 Pro back triple camera 4.25mm f/1.8"
}]
```

And `2023-05-02_21-33-15_IMG_6177.MOV` is the following:
```json
[{
  "SourceFile": "originals/2023/05/20230503_043315_41744952.mov",
  "ExifToolVersion": 12.56,
  "FileName": "20230503_043315_41744952.mov",
  "Directory": "originals/2023/05",
  "FileSize": "1531 kB",
  "FileModifyDate": "2023:08:27 04:24:09-07:00",
  "FileAccessDate": "2023:08:27 04:24:09-07:00",
  "FileInodeChangeDate": "2023:08:27 16:01:31-07:00",
  "FilePermissions": "-rw-rw-r--",
  "FileType": "MOV",
  "FileTypeExtension": "mov",
  "MIMEType": "video/quicktime",
  "MajorBrand": "Apple QuickTime (.MOV/QT)",
  "MinorVersion": "0.0.0",
  "CompatibleBrands": ["qt  "],
  "MediaDataSize": 1524611,
  "MediaDataOffset": 36,
  "MovieHeaderVersion": 0,
  "CreateDate": "2023:05:04 05:50:16",
  "ModifyDate": "2023:05:04 05:50:16",
  "TimeScale": 600,
  "Duration": "2.64 s",
  "PreferredRate": 1,
  "PreferredVolume": "100.00%",
  "PreviewTime": "0 s",
  "PreviewDuration": "0 s",
  "PosterTime": "0 s",
  "SelectionTime": "0 s",
  "SelectionDuration": "0 s",
  "CurrentTime": "0 s",
  "NextTrackID": 6,
  "TrackHeaderVersion": 0,
  "TrackCreateDate": "2023:05:04 05:50:16",
  "TrackModifyDate": "2023:05:04 05:50:16",
  "TrackID": 1,
  "TrackDuration": "2.64 s",
  "TrackLayer": 0,
  "TrackVolume": "100.00%",
  "Balance": 0,
  "AudioFormat": "mp4a",
  "AudioChannels": 1,
  "AudioBitsPerSample": 16,
  "AudioSampleRate": 44100,
  "PurchaseFileFormat": "mp4a",
  "ImageWidth": 960,
  "ImageHeight": 720,
  "CleanApertureDimensions": "872x654",
  "ProductionApertureDimensions": "960x720",
  "EncodedPixelsDimensions": "960x720",
  "GraphicsMode": "ditherCopy",
  "OpColor": "32768 32768 32768",
  "CompressorID": "avc1",
  "SourceImageWidth": 960,
  "SourceImageHeight": 720,
  "XResolution": 72,
  "YResolution": 72,
  "CompressorName": "H.264",
  "BitDepth": 24,
  "VideoFrameRate": 14.006,
  "Warning": "[minor] The ExtractEmbedded option may find more tags in the media data",
  "MatrixStructure": "1 0 0 0 1 0 0 0 1",
  "ContentDescribes": "Track 2",
  "MediaHeaderVersion": 0,
  "MediaCreateDate": "2023:05:04 05:50:16",
  "MediaModifyDate": "2023:05:04 05:50:16",
  "MediaTimeScale": 600,
  "MediaDuration": "0.00 s",
  "MediaLanguageCode": "und",
  "GenMediaVersion": 0,
  "GenFlags": "0 0 0",
  "GenGraphicsMode": "ditherCopy",
  "GenOpColor": "32768 32768 32768",
  "GenBalance": 0,
  "HandlerClass": "Data Handler",
  "HandlerVendorID": "Apple",
  "HandlerDescription": "Core Media Data Handler",
  "MetaFormat": "mebx",
  "HandlerType": "Metadata Tags",
  "LivePhotoAuto": 1,
  "ContentIdentifier": "CB21A521-567B-4579-B045-B29CD2474CB3",
  "LivePhotoVitalityScore": 1,
  "LivePhotoVitalityScoringVersion": 4,
  "Make": "Apple",
  "Model": "iPhone 13",
  "Software": "16.4.1",
  "CreationDate": "2023:05:02 21:33:15-07:00",
  "ImageSize": "960x720",
  "Megapixels": 0.691,
  "AvgBitrate": "4.62 Mbps",
  "Rotation": 90
}]
```

I can confirm that they are still in a stack.
![image](https://github.com/photoprism/photoprism/assets/12904259/aaead799-9631-446b-a2e1-831e81f75e3c)

**Reply by lastzero** [reply]
Note that already stacked files will not be unstacked automatically, even if you later change the settings. If you believe that this is a bug that (also) occurs when you newly index or import these files (it seems you originally imported them), you can send us file samples for testing:
- https://www.photoprism.app/contact#file-samples

**Reply by graciousgrey** [reply]
I just tested importing files with the names: 2021-07-16_12-43-54_IMG_6177.heic and 2023-05-02_21-33-15_IMG_6177.mov and they did not get stacked.

**Reply by Abejo** [reply]
Is there any solution or way to disable this option? I have a lot of photos, especially the scanned ones that I have stacked and they are different photographs.

**Reply by lastzero** [reply]
To answer this question, we first need to find out why they were stacked in the first place...

**Reply by lastzero** [reply]
From what I can see in the screenshot you shared, there are additional files in the stack, including a json file. That makes me think the metadata of the json file could have been applied to all / multiple files, so that the exiftool output for the two files you shared doesn't tell the whole truth...

**Reply by Abejo** [reply]
There is no json, they are just jpg (I leave you 2 of the photos that were stacked and a screenshot of the directory). I started to disassemble the batteries manually, but I would like to know if there is a way to disassemble them all since there are quite a few photos.
![Screenshot_20231218_161412](https://github.com/photoprism/photoprism/assets/6800062/bd3ff72a-c687-4184-b50e-5e885f8a671e)
![197811_F01](https://github.com/photoprism/photoprism/assets/6800062/4b137346-9b62-4e18-9250-86a521bbc4db)
![197811_F02](https://github.com/photoprism/photoprism/assets/6800062/4b58150a-7621-4e91-9351-f47bdb1f3b2b)

**Reply by lastzero** [reply]
> There is no json
![json](https://github.com/photoprism/photoprism/assets/301686/68803d0d-4e09-47d2-b3fd-fdf3df402eff)

**Reply by Abejo** [reply]
But that is not a capture of the photos I sent 🤔

**Reply by lastzero** [reply]
No, but of the user interface showing the files. Isn't that a screenshot you created?

**Reply by Abejo** [reply]
Yes, but that screenshot is not the one I sent, the dates of my files are not those, I do not have any mov files as that image shows

**Reply by ALEEF02** [reply]
I don't know who @Abejo is and his photos are unrelated to mine, effectively clogging this thread. I have attached my files in an email, @lastzero.

**Reply by graciousgrey** [reply]
@ALEEF02 I just answered your mail :)

@Abejo I had a look at your files. Those are stacked because they share the exact same place and time. If you disable stacking by place and time in settings in the future files won't be stacked for this reason anymore. However this will not unstack already stacked files. A unstack all button is planned for the future but not implemented yet.

**Comment by Maximus48p** [comment]
I have also similar issues. Over 17000 photo's and a lot of them are stacked wrong, most of them are jpg stacked with a completely different mov file. Undoable to unstack them one by one. 
I would like to unstack them all as bulk change. Have been browsing but can't find a solution in the documentation or forums.
Any suggestions? Thank you very much.

**Reply by graciousgrey** [reply]
There is no bulk unstack option available yet. This is because unstacking is a resource-intensive operation that requires each file to be re-indexed.

The result also depends on the exact order in which you unstack the files, as non-media sidecar files, for example, remain bound to the remaining media file in a stack. We consider providing a command for this in a future release and appreciate [any contributions](https://docs.photoprism.app/developer-guide/) in this regard.

Note that it is not possible to disable stacking of files with the same name as this would break important functionality, most notably support for Apple [Live Photos](https://docs.photoprism.app/user-guide/organize/video/#live-photos) (which consist of a photo and a video file), any other multi-file/hybrid formats like RAW/JPEG, and indexing of metadata from XMP/JSON sidecar files.

If your camera uses the exact same name for different files you might want to put those in different folders or use a tool like PhotoSync to import files that allows you to define a custom naming.

**Reply by Maximus48p** [reply]
Thank you for your answer. My comments on this:
- A specific function for this to unstack in bulk would be great. And if i may give some suggestions :).... Hopefully including some parameters. For example;
- - between certain dates
- - in case of stack of two, unstack where date file 1 is different of date file 2.
- - in case of stack over two files, unstack only the file who has a different date.
- - in case where camera is different
-  or better first to create a specific selection first. For example; Show all stacked photos where...
- - camera/model is different
- - create dates are different
- - file dates are different
- - include a mov and jpg file
- - more then 2 files are stacked
- - only from certain album
- - not having metadata
In this case you can make different selections first and put them first in a temporary album. You can do a second selection on only these foto's. In this way you can unstack them in a controlled way using several steps.

- I would love to contribute, but i'm a noob, sorry. I can only help with suggestions.

Back to my issue:
- I have importing photo from the last decade and they are taken with multiple different camera's and phones. So it's completely normal in this situation that filenames are double. 

What to do next with so many wrong stacked photo's? 
Do i have to start all over again and import 17000+ photo's again with stacking option disabled? 
I think i need to cry...
Is there no other solution? Please, let there be some other solution...

**Reply by lastzero** [reply]
> I have importing photo from the last decade and they are taken with multiple different camera's and phones. So it's completely normal in this situation that filenames are double.

It is not a problem when there are duplicate file names. However, in this case the files should not be in the same directory unless they belong together. I don't understand how you were able to put all the camera files (which typically are in the format IMG_1234.JPG) in one folder without overwriting existing files, since that only allows for 10000 different names?
